test/components/Home/KeyFeatures.test.jsx

```diff
@@ -7,19 +7,19 @@ import createMatchMedia from '../../helpers/createMatchMedia';
 const MockKeyFeaturesDefault = () => <KeyFeatures description="Example Text" />;
 const MockKeyFeaturesMobile = () => <KeyFeatures isReallySmallScreen description="Example Text" />;
 
-it('renders 67% width default', () => {
+it('renders 85% width default', () => {
```

Comment:
This is fine, but tests tied to specific CSS values like this probably won't be very resilient. I don't think we'd want this test to fail if someone decides that actually 84% is better for desktop than 85%. We may want to associate the test with more significant values/ranges.  Here it looks like mobile always renders `100%`, which seems like a pretty set value. So something like `expect(descriptionStyles.width).not.toBe('100%')` may be more reliable.

In general we may be able to make tests like 'It renders desktop display on desktop' which checks for a `desktopLayout` class, and then a 'It renders mobile display on mobile' which checks for a `mobileLayout` class. Then we can update the precise details of the class as we need without causing the tests to fail.

Reply:
Fully agreed and something discussed in depth in our discord DM. 

Changed all to have a base value so .toBe(basevalue) & .not.toBe(basevalue)

This is SLIGHTLY more resilient, but not by much as when base value changes you have the same effect. 
I think all CSS value based tests are no good.  Maybe adjusting tests to be more resilient is a future issue for the code base?

Reply:
I think the bigger question is what exactly is this supposed to be testing? If we are testing whether or not our styles change at certain values, I think it would make more sense to create a function that tests whether this does what it's supposed to to set `isReallySmallScreen` in `src/components/Messages/MessageButtonGroup.jsx`:

```ts
  const isReallySmallScreen = useMediaQuery('(max-width: 480px)');
```

If we can test `useMediaQuery`, then we won't really need to use these tests, will we?

Reply:
I agree that they are very temperamental. Since they are directly coupled with hard coded values, it will either be annoying or ignored by future devs, which kind of invalidates the purpose of tests, which is an alarm for unexpected changes.

If we can test *behaviors* for the user, then that will be more resilient. And I think will also let us ask whether or not a test is useful, but that is only my opinion.

Reply:
I'm not sure I follow 100%, but I also believe that testing that react is doing react is silly and unnecessary. The maintainers of react and MUI are testing them, we don't need to as well.  I don't think we need to test if `useMediaQuery` works either. Kind of the point of using a library. 

I feel like the behavior (testing onclicks and such) should be done.

I'm not sold on checking css/styling or if mobile component is rendering in mobile screen or if desktop component is rendering in desktop screen.  It will render whatever we tell it to render. 

I'm not sure, I'm still studying.